crates/bevy_render/src/texture/image.rs

```diff
@@ -160,6 +160,25 @@ impl ImageSampler {
     }
 }
 
+/// Proxy type used to implement [`Serialize`] and [`Deserialize`] for [`ImageSampler`].
+#[derive(Serialize, Deserialize)]
+#[serde(remote = "wgpu::SamplerDescriptor<'static>")]
+struct WgpuSamplerDescriptor {
+    #[serde(skip)]
+    label: wgpu::Label<'static>,
+    address_mode_u: wgpu::AddressMode,
+    address_mode_v: wgpu::AddressMode,
+    address_mode_w: wgpu::AddressMode,
+    mag_filter: wgpu::FilterMode,
+    min_filter: wgpu::FilterMode,
+    mipmap_filter: wgpu::FilterMode,
+    lod_min_clamp: f32,
+    lod_max_clamp: f32,
+    compare: Option<wgpu::CompareFunction>,
+    anisotropy_clamp: u16,
+    border_color: Option<wgpu::SamplerBorderColor>,
+}
+
```

Comment:
Can we newtype the wgpu SamplerDescriptor type instead of reimplementing it?

Reply:
From my understanding, that's not possible using serde derive, a manual `Serialize`/`Deserialize` impl would need to be added for `ImageSampler`. Is that preferable ?

Reply:
Trying to think what would be most maintainable. I am concerned about desync.

Reply:
i think this is fine - [At compile time, Serde will check that all the fields in the definition you provided match the fields in the remote type](https://serde.rs/remote-derive.html)